service/lib/dinstaller/storage/dasd/format_operation.rb

```diff
@@ -27,37 +27,33 @@ module DInstaller
   module Storage
     module DASD
       # Operation to format the given set of DASDs
-      #
-      # TODO: we plan to change the approach to DASD formatting, so this class will change
-      # very soon.
       class FormatOperation
         # Constructor
         #
         # @param dasds [Array<Y2S390:Dasd>] devices to format
-        def initialize(dasds)
+        def initialize(dasds, on_progress = [], on_finish = [])
           @process = Y2S390::FormatProcess.new(dasds)
+          @on_progress = on_progress
+          @on_finish = on_finish
         end
 
-        # Formats all the given DASDs
+        # Starts a format process on the given DASDs
         #
-        # NOTE: this algorithm is pretty much copied from Y2S390::Dialogs::FormatDialog
+        # NOTE: Does a device need to be reactivated after formating it? The code at
+        # DasdActions::Activate seems to imply that, but the code at DasdActions::Format contains a
+        # comment stating otherwise. Let's do nothing for the time being.
         #
-        # @return [Boolean] true if the operation succeeds for all the DASDs
+        # @return [Array<Y2S390::FormatStatus>, nil] Initial status for all DASDs, nil if the format
+        #   process couldn't be started.
         def run
-          # NOTE: Does a device need to be reactivated after formating it?
-          #       The code at DasdActions::Activate seems to imply that
-          #       But the code at DasdActions::Format contains a comment stating otherwise
-          return false unless start?
+          return nil unless start?
 
           process.initialize_summary
-          while process.running?
-            process.update_summary
-            # TODO: trigger a callback to notify the process status information
-            sleep(0.2)
+          Thread.new do
```

Comment:
Oh, a thread.

ruby-dbus is not thread safe if the threads are sharing a connection.

We need to prove that threads will not collide on the bus, maybe add synchronization *somewhere*